lnwire/onion_error.go

```diff
@@ -1271,14 +1272,19 @@ func (f *FailInvalidBlinding) Encode(w *bytes.Buffer, _ uint32) error {
 }
 
 // NewInvalidBlinding creates new instance of FailInvalidBlinding.
-func NewInvalidBlinding(onion []byte) *FailInvalidBlinding {
+func NewInvalidBlinding(
+	onion fn.Option[[OnionPacketSize]byte]) *FailInvalidBlinding {
 	// The spec allows empty onion hashes for invalid blinding, so we only
 	// include our onion hash if it's provided.
-	if onion == nil {
+	if onion.IsNone() {
 		return &FailInvalidBlinding{}
 	}
 
-	return &FailInvalidBlinding{OnionSHA256: sha256.Sum256(onion)}
+	shaSum := fn.MapOptionZ(onion, func(o [OnionPacketSize]byte) [32]byte {
+		return sha256.Sum256(o[:])
```

Comment:
the `onion` can be an empty byte array here, pre-existing tho, worst case we hash over zeroed byte array.

Reply:
from what I can tell, arrays are always fixed size so they can't be "empty". If it is a zeroed byte array we will hash the zeroed array here, yes. Is this a problem?